src/components/services/UptimeKuma.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <Generic :item="item">
+    <template #content>
+      <p class="title is-4">{{ item.name }}</p>
+      <p class="subtitle is-6">
+        <template v-if="item.subtitle">
+          {{ item.subtitle }}
+        </template>
+        <template v-else-if="status">
+          {{ statusMessage }}
+        </template>
+      </p>
+    </template>
+    <template #indicator>
+      <div v-if="status" class="status" :class="status">
+        {{ uptime }}&percnt;
+      </div>
+    </template>
+  </Generic>
+</template>
+
+<script>
+import service from "@/mixins/service.js";
+import Generic from "./Generic.vue";
+
+export default {
+  name: "UptimeKuma",
+  mixins: [service],
+  props: {
+    item: Object,
+  },
+  components: {
+    Generic,
+  },
+  data: () => ({
+    incident: null,
+    heartbeat: null,
+  }),
+  computed: {
+    status: function () {
+      if (!this.incident) {
+        return "";
+      }
+      return this.incident.ok && this.incident.incident == null
+        ? "good"
+        : "bad";
+    },
+    statusMessage: function () {
+      if (!this.incident) {
+        return "";
+      }
+      const inc = this.incident.incident;
+      return inc ? inc.title : "No Incidents";
+    },
+    uptime: function () {
+      if (!this.heartbeat) {
+        return 0;
+      }
+      const data = Object.values(this.heartbeat.uptimeList);
+      const percent = data.reduce((a, b) => a + b, 0) / data.length || 0;
+      return (percent * 100).toFixed(1);
+    },
+  },
+  created() {
+    this.fetchStatus();
+  },
+  methods: {
+    fetchStatus: async function () {
+      const incident = await this.fetch("/api/status-page/incident").catch(
+        (e) => console.log(e)
+      );
+      this.incident = incident;
+
+      const heartbeat = await this.fetch("/api/status-page/heartbeat").catch(
+        (e) => console.log(e)
+      );
+      this.heartbeat = heartbeat;
```

Comment:
Could you use a `.then()` instead of `await` ? That way both requests can run in parallel instead of one after the other.